shared/protocol/src/main/java/io/pravega/shared/segment/StreamSegmentNameUtils.java

```diff
@@ -366,4 +375,12 @@ public static boolean isTableSegment(String qualifiedName) {
         return tokens[1].equals(TABLES);
     }
     // endregion
+    
+    // region watermark
+    public static String getMarkSegmentForStream(String stream) {
+        StringBuffer sb = new StringBuffer();
+        sb.append(MARK_PREFIX);
+        sb.append(stream);
```

Comment:
We should have a precondition that `stream` does not start with the mark_prefix already as this makes little sense. (It probably also doesn't make sense for reader groups)

Reply:
That check should be in stream manager interface perhaps. Because controller is used to create both internal and external streams and it wouldnt know where the request is originating from.